### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by iratxe-zunzunegui**
I uploaded the files corresponding to the Assignment 4 to a folder with my username. Unfortunately, I'm not sure where I can find my Student ID.

### CODE REVIEW

**Comment by iratxe-zunzunegui** [comment]
I added the tasks 8 and 9.

File: Assignment4/iratxe-zunzunegui/task07.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+ns = Namespace("http://somewhere#")
+
```
Comment (SerPablo) [inline Assignment4/iratxe-zunzunegui/task07.py:28]: Do it recursively for sub-subclasses

File: Assignment4/iratxe-zunzunegui/task07.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  print(r.p, r.o)
+
+"""**TASK 7.4:  List the name of the persons who know Rocky**"""
+
+# TO DO -- because it doesn't specify doing the exercise with both RDFLib and SPARQL, we will do it just with SPARQL
+
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0/")
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+from rdflib import XSD
+
```
Comment (SerPablo) [inline Assignment4/iratxe-zunzunegui/task07.py:122]: Only persons

**Review by SerPablo — COMMENTED** [review]
Review comments for task 7, remove files for task 8 and 9 for this pull request, try 7.4 and 7.5 with rdflib

File: Assignment4/iratxe-zunzunegui/task07.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+)
+
+# Visualize the results
+
+for r in g.query(q4, initBindings = {'?RockyFullName' : Literal('Rocky Smith', datatype=XSD.string)}):
+  print(r.Given)
+
+"""**Task 7.5: List the entities who know at least two other entities in the graph**"""
+
+# TO DO -- because it doesn't specify doing the exercise with both RDFLib and SPARQL, we will do it just with SPARQL
```
Comment (dgarijo) [inline Assignment4/iratxe-zunzunegui/task07.py:140]: do it also with rdflib please

File: Assignment4/iratxe-zunzunegui/task07.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+# TO DO
+
+##RDFlib
+print("RDFlib result:")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+    print(s)
+
+for s,p,o in g.triples ((None, RDFS.subClassOf, ns.Person)):
+    for s1,p1,o1 in g.triples((None, RDF.type, s)):
+        print (s1)
```
Comment (dgarijo) [inline Assignment4/iratxe-zunzunegui/task07.py:63]: same issue as above

**Comment by dgarijo** [comment]
Last commits still do not address the main issues detected

**Pull request merged**